From baea9c46be3efde935e91ad0d78f6e5086315346 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 28 Jul 2017 17:43:01 +0800
Subject: [PATCH 2313/5242] MLK-16013-34 staging: typec: Support cc changes on
 RP pull-up value

commit  24da25413402ba47baeaa3e39176ee0ac950f4ff from
https://source.codeaurora.org/external/imx/linux-imx.git

CC line changes on souce side RP value is permitted per standard, in
this case we should move to SNK_ATTACH_WAIT.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index 82e0448..1924d42 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -3618,6 +3618,9 @@ static void _tcpm_cc_change(struct tcpm_port *port, enum typec_cc_status cc1,
 			new_state = SNK_UNATTACHED;
 		else if (port->vbus_present)
 			new_state = tcpm_try_src(port) ? SRC_TRY : SNK_ATTACHED;
+		else if (cc1 > TYPEC_CC_RP_DEF || cc2 > TYPEC_CC_RP_DEF)
+			/* CC changes on pull-up value */
+			new_state = SNK_ATTACH_WAIT;
 		else
 			new_state = SNK_UNATTACHED;
 		if (new_state != port->delayed_state)
-- 
1.7.9.5

